mlir/include/mlir/Dialect/OpenMP/OpenMPOps.td

```diff
@@ -1203,9 +1201,11 @@ def MapInfoOp : OpenMP_Op<"map_info", [AttrSizedOperandSegments]> {
   }];
 
   let assemblyFormat = [{
-    `var_ptr` `(` $var_ptr `:` type($var_ptr) `,` $var_type`)`
     oilist(
-        `var_ptr_ptr` `(` $var_ptr_ptr `:` type($var_ptr_ptr) `)`
+        `val` `(` $val `:` type($val) `)`
+      | `var_ptr` `(` $var_ptr `:` type($var_ptr) `)`
+      | `var_type` `(` $var_type `)`
+      | `var_ptr_ptr` `(` $var_ptr_ptr `:` type($var_ptr_ptr) `)`
```

Comment:
`$var_type` is only required for Reference/Pointer types, if the `value` is not a reference, then it need not be present.

Reply:
Currently, `$var_type` is being set to the element_type for `$var_ptr`, or the type of the val for `$val`.

This is done so that later we can always just call a single `get_var_type()` to get the type, instead of having to check if it is a `$val` or `$var_ptr` and then using appropriate functions to retrieve the type.

Do you prefer the latter approach?

Reply:
Thanks, I've updated this.